app/src/main/java/eu/kanade/tachiyomi/ui/setting/SettingsDownloadController.kt

```diff
@@ -76,26 +78,74 @@ class SettingsDownloadController : SettingsController() {
             }
         }
 
-        val dbCategories = db.getCategories().executeAsBlocking()
-        val categories = listOf(Category.createDefault(context)) + dbCategories
-
         preferenceCategory {
```

Comment:
I'm not too sure about all these changes tbh
1. I agree the setting `download only completely read` is not particularly useful most of the time except
   - For example I like to deactivate all the library restrictions to update all my library once in a while (like 2 times a month) 
   - When you update a particular manga via manga_details, you don't really want to download all the chapters if not necessary
2.  I'm not sure I understand the difference between the `Download chapters after reading` and `Download ahead if chapter is not downloaded`. Shouldn't one of them be a switch ?
3. I kind of like the setting to limit the number of downloaded chapters for new chapters (was the same as download while reading for now), some sources like to mass release chapters and I don't think you really want to download everything at once (especially to ease source traffic)

Reply:
1. The issue isnt really the setting itself but more the fact that what people always say: Tachiyomi has way too many options for a reading app. I get your use case for it but I dont think I want to add more complexity to the update/download restrictions just yet
2. I was trying to think of the wording of them, but it goes something like this
   * `Download ahead if chapter is not downloaded`: This is the switch option basically, if its never, the auto dowload is disabled. If it's on, depending on the number set it will start auto downloading. So if set the setting to 2, you finish reading chapter 3, and chapter 5 isn't downloaded it will download that chapter and more based on...
   * `Download chapters after reading` (this setting will be hidden if the above is off): This decides how many more chapters along with the first non-downloaded chapter to download. If this is set to 3 for example, it would download 5, 6, and 7. 
3. Same as point 1. I will say you should make a PR for point 1 and 2 to the main tachi repo and ill pull it back if they accept. It seems like a feature more for source damage control so it would make sense to ask them if they think the setting would make sense than me

Reply:
1. Well I agree, it's far from necessary, I'll remove it
2. It's more clear but I don't really see the point of having 2 settings like that. As you said too many/complex settings is not good. Wouldn't it be enough to just set how many chapters you want downloaded ahead ?

If I remove the `auto-download only over wifi` and `download only completely read` settings and only keep (bold is what I add) :
- **Download after reading** ? `switch` (to avoid migration, as you said)
- Download new chapters ? `switch`
- **How many downloaded chapters you want per manga** (= auto-download restrictions, may need a better name) `intList`
- Categories for new chapters `triStateList`

Imo it seems simple enough while adding some useful features that would answer 95% of the cases. What do you think ?
We can even put all of that in an `Auto-download` category like main.

Reply:
I kinda agree with point 2, but to a degree. I'd still say keep all the existing settings the same and add a single setting for download ahead/auto download and thats
"Download next unread chapters after reading": Never/Disabled, Next chapter, next 2, 3, 5
This we can use the logic to check the next _x_ chapter(s) and download them if they aren't (assuming the currently read chapter was downloaded. So yeah much more simple.

so just:
* Download new chapters
* Categories

* "Download ahead" (section could be without a title since its the one setting)
   * "Download next unread chapters after reading"

And like i said for previous point 3 maybe just make a pr to main for a download new chapters limit

Reply:
Yeah seems good like that ! I'll go with your first option since a single section could indeed be quite confusing
I'll implement/modify all that during the week
And alright I'll see to make a PR to main for the new chapters limit